logging/net_appender.go

```diff
@@ -0,0 +1,324 @@
+package logging
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap/zapcore"
+	apppb "go.viam.com/api/app/v1"
+	"go.viam.com/utils"
+	"go.viam.com/utils/protoutils"
+	"go.viam.com/utils/rpc"
+	"google.golang.org/protobuf/types/known/structpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+const (
+	defaultMaxQueueSize = 20000
+	writeBatchSize      = 100
+)
+
+// CloudConfig contains the necessary inputs to send logs to the app backend over grpc.
+type CloudConfig struct {
+	AppAddress string
+	ID         string
+	Secret     string
+}
+
+// NewNetAppender creates a NetAppender to send log events to the app backend.
+func NewNetAppender(config *CloudConfig) (*NetAppender, error) {
```

Comment:
I changed the constructor to not require a logger (for simplicity without loss of generality). Additionally I removed the log level as "appenders" don't concern themselves on whether to log.